src/components/research-form-wizard/ResearchFormWizard.tsx

```diff
@@ -0,0 +1,161 @@
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+import Modal from 'react-bootstrap/Modal';
+import styled from 'styled-components';
+
+import { useCookiePreferences } from '@/hooks/useCookiePreferences';
+import { useResearchWizardEvents } from '@/hooks/useResearchWizardEvents';
+import { useResearchWizardStore } from '@/stores/researchWizard';
+import { recordResearchFromEvent } from '@/utils/analytics';
+
+import { StepLayout } from './StepLayout';
+import ThumbsUpIcon from './thumbs-up.svg';
+
+const MobileWrapper = styled.div`
+  width: 276px;
+  height: 72px;
+  position: fixed;
+  bottom: 20px;
+  left: 50%;
+  transform: translateX(-50%);
+  padding: 10px 20px;
+  background-color: white;
+  color: #000000;
+  border: none;
+  border-radius: 100px;
+  cursor: pointer;
+  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
+  display: flex;
+  align-items: center;
+  gap: 15px;
+`;
+
+const MobileFormWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const DesktopFormWrapper = styled.div`
+  position: fixed;
+  bottom: 1em;
+  right: 1em;
+  margin: 0 auto;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-top: 10px;
+  border-radius: 50%;
+  width: 48px;
+  height: 48px;
+  color: white;
+`;
+
+export const ResearchFormWizard = () => {
+  const set = useResearchWizardStore((state) => state.set);
+  const [matches, setMatches] = useState(true);
+  const currentStepIndex = useResearchWizardStore((state) => state.currentStepIndex);
+  const formSteps = useResearchWizardStore((state) => state.formSteps);
+  const currentStepSubmission = useResearchWizardStore((state) => state.currentStepSubmission);
+  const cookieData = useCookiePreferences();
+  const { showMobileResearchForm, setShowMobileResearchForm, isResearchFormDismissed, setIsResearchFormDismissed } =
+    useResearchWizardEvents();
+  const [researchEventSent, setResearchEventSent] = useState(false);
+  const [currentStep, setCurrentStep] = useState<JSX.Element>(formSteps[currentStepIndex].component());
+
+  useEffect(() => {
+    setMatches(window.matchMedia('(min-width: 1120px)').matches);
+  }, []);
+
+  useEffect(() => {
+    window.matchMedia('(min-width: 1120px)').addEventListener('change', (e) => setMatches(e.matches));
+  }, []);
+
+  const dismissForm = () => {
+    recordResearchFromEvent('close-research-form', { questionNumber: currentStepIndex + 1 });
```

Comment:
@jackson-harris-iii Did Jo or Spencer recommend this structure? Recording an index would require a look-up against some other data store to resolve what the action question was.  Why not require the full info such as `"event": "q3_developer_survey", "properties": {  "question": "Which of the following best describes you..."  "answer": "Other - wen 🚀 ?"}`

Reply:
Or perhaps we are recording the full question somewhere and I have missed it.

Reply:
The dismiss is when the user clicks the x, this allows us to record that a user is dismissing the form.  We record interactions with the question information and their answers in 'handleFormButton', this function fires off the submission on each click of the next or submit button :

`const handleFormButton = () => {
    if (currentStepSubmission) {
      recordResearchFromEvent('q3-developer-survey-submission', {
        questionNumber: currentStepIndex + 1,
        response: currentStepSubmission,
      });

      if (currentStepIndex < formSteps.length - 1) {
        setCurrentStep(formSteps[currentStepIndex + 1].component());
        useResearchWizardStore.setState({ currentStepIndex: currentStepIndex + 1 });
      } else {
        setIsResearchFormDismissed(true);
        localStorage.setItem('researchFormDismissed', 'true');
      }
      set({ currentStepSubmission: '' });
    }
  };`

Reply:
https://github.com/near/near-discovery/pull/1251/files#diff-48f525b8d86b2469507944329205ec412edc43cc9a3e5f4ac11af7408d153eebR89